src/tools/business/gcal_tool.py

```
@@ -164,11 +220,24 @@ def _normalize_datetime_to_calendar_tz(
+        base: Dict[str, Any] = {}
+        overlay: Dict[str, Any] = {}
         if context and getattr(context, "get_config_value", None):
-            return context.get_config_value("tools.google_calendar", {}) or {}
-        return self._load_config()
+            base = context.get_config_value("tools.google_calendar", {}) or {}
+            ctx_name = getattr(context, "context_name", None)
+            if ctx_name:
+                # Per-context override under contexts.<name>.tool_overrides.google_calendar (avoids colliding with tools list)
+                try:
+                    overlay = context.get_config_value(f"contexts.{ctx_name}.tool_overrides.google_calendar", {}) or {}
+                except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)